packages/blockchain-api/src/cacheDataSource.ts

```diff
@@ -0,0 +1,22 @@
+import { DataSource, DataSourceConfig } from 'apollo-datasource'
+import { KeyValueCache } from 'apollo-server-caching'
+
+export default class CacheDataSource<T> extends DataSource {
+  // @ts-ignore We can't express that initialize will definitely be called
```

Comment:
If we set it in the constructor instead, would that solve the TS error?

Reply:
The way apollo-data sources are setup is that they receive the cache object in the `initialize` function unfortunately